# SQL-Analyzing-Customer-Transaction-Data-to-Identify-Spending-Patterns-and-Optimize-Product-Offerings

In [1]:
!jt -t grade3


In [2]:
import pandas as pd
import sqlite3

In [3]:
df=pd.read_csv("retail_superstore.csv")

In [4]:
df.head()

Row ID        Order_ID  Order_Date   Ship_Date       Ship_Mode Customer_ID  \
0       1  CA-2017-152156  08-11-2017  11-11-2017    Second Class    CG-12520   
1       2  CA-2017-152156  08-11-2017  11-11-2017    Second Class    CG-12520   
2       3  CA-2017-138688  12-06-2017  16-06-2017    Second Class    DV-13045   
3       4  US-2016-108966  11-10-2016  18-10-2016  Standard Class    SO-20335   
4       5  US-2016-108966  11-10-2016  18-10-2016  Standard Class    SO-20335   

     Customer_Name    Segment        Country             City       State  \
0      Claire Gute   Consumer  United States        Henderson    Kentucky   
1      Claire Gute   Consumer  United States        Henderson    Kentucky   
2  Darrin Van Huff  Corporate  United States      Los Angeles  California   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   

   Postal_Code Region       Product_ID         Category Sub_Category  \
0      42420.0  South  FUR-BO-10001798        Furniture    Bookcases   
1      42420.0  South  FUR-CH-10000454        Furniture       Chairs   
2      90036.0   West  OFF-LA-10000240  Office Supplies       Labels   
3      33311.0  South  FUR-TA-10000577        Furniture       Tables   
4      33311.0  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product_Name     Sales  
0                  Bush Somerset Collection Bookcase  261.9600  
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400  
2  Self-Adhesive Address Labels for Typewriters b...   14.6200  
3      Bretford CR4500 Series Slim Rectangular Table  957.5775  
4                     Eldon Fold 'N Roll Cart System   22.3680

# Table Of Contents
* [Project Description: Superstore Sales Analysis](#project_description)
    * [Objectives](#objectives)
    * [Dataset Overview](#dataset_overview)
    * [Expected Outcomes](#expected_outcomes)
* [Customer And Order Analysis](#customer_and_order_analysis)
* [Product Performance](#product_performance)
* [Geographic Analysis](#geographic_analysis)
* [Seasonal and Time-Based Analysis](#i1)


# Project Description: Superstore Sales Analysis <a class="anchor" id="project_description"></a>


"""This project involves analyzing transaction data from a retail superstore to identify 
customer spending patterns, optimize product offerings, and improve business strategies. 
The dataset contains comprehensive sales and customer information, including details about each order, 
customer demographics, product categories, and regional data. By exploring this data, we aim to gain 
actionable insights to help the superstore enhance customer satisfaction, increase sales, 
and manage inventory more effectively."""

# Objectives <a id="objectives"></a>

# Dataset Overview <a id="dataset_overview"></a>

# Expected Outcomes <a id="expected_outcomes"></a>


In [5]:
conn=sqlite3.connect(':memory:')
df.to_sql("retail_superstore",conn,index=True)

9800

In [6]:
# Example SQL query to select data
query = "SELECT * FROM retail_superstore"
result = pd.read_sql_query(query, conn)
result

index  Row ID        Order_ID  Order_Date   Ship_Date       Ship_Mode  \
0         0       1  CA-2017-152156  08-11-2017  11-11-2017    Second Class   
1         1       2  CA-2017-152156  08-11-2017  11-11-2017    Second Class   
2         2       3  CA-2017-138688  12-06-2017  16-06-2017    Second Class   
3         3       4  US-2016-108966  11-10-2016  18-10-2016  Standard Class   
4         4       5  US-2016-108966  11-10-2016  18-10-2016  Standard Class   
...     ...     ...             ...         ...         ...             ...   
9795   9795    9796  CA-2017-125920  21-05-2017  28-05-2017  Standard Class   
9796   9796    9797  CA-2016-128608  12-01-2016  17-01-2016  Standard Class   
9797   9797    9798  CA-2016-128608  12-01-2016  17-01-2016  Standard Class   
9798   9798    9799  CA-2016-128608  12-01-2016  17-01-2016  Standard Class   
9799   9799    9800  CA-2016-128608  12-01-2016  17-01-2016  Standard Class   

     Customer_ID     Customer_Name    Segment        Country             City  \
0       CG-12520       Claire Gute   Consumer  United States        Henderson   
1       CG-12520       Claire Gute   Consumer  United States        Henderson   
2       DV-13045   Darrin Van Huff  Corporate  United States      Los Angeles   
3       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
4       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
...          ...               ...        ...            ...              ...   
9795    SH-19975     Sally Hughsby  Corporate  United States          Chicago   
9796    CS-12490  Cindy Schnelling  Corporate  United States           Toledo   
9797    CS-12490  Cindy Schnelling  Corporate  United States           Toledo   
9798    CS-12490  Cindy Schnelling  Corporate  United States           Toledo   
9799    CS-12490  Cindy Schnelling  Corporate  United States           Toledo   

           State  Postal_Code   Region       Product_ID         Category  \
0       Kentucky      42420.0    South  FUR-BO-10001798        Furniture   
1       Kentucky      42420.0    South  FUR-CH-10000454        Furniture   
2     California      90036.0     West  OFF-LA-10000240  Office Supplies   
3        Florida      33311.0    South  FUR-TA-10000577        Furniture   
4        Florida      33311.0    South  OFF-ST-10000760  Office Supplies   
...          ...          ...      ...              ...              ...   
9795    Illinois      60610.0  Central  OFF-BI-10003429  Office Supplies   
9796        Ohio      43615.0     East  OFF-AR-10001374  Office Supplies   
9797        Ohio      43615.0     East  TEC-PH-10004977       Technology   
9798        Ohio      43615.0     East  TEC-PH-10000912       Technology   
9799        Ohio      43615.0     East  TEC-AC-10000487       Technology   

     Sub_Category                                       Product_Name     Sales  
0       Bookcases                  Bush Somerset Collection Bookcase  261.9600  
1          Chairs  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400  
2          Labels  Self-Adhesive Address Labels for Typewriters b...   14.6200  
3          Tables      Bretford CR4500 Series Slim Rectangular Table  957.5775  
4         Storage                     Eldon Fold 'N Roll Cart System   22.3680  
...           ...                                                ...       ...  
9795      Binders  Cardinal HOLDit! Binder Insert Strips,Extra St...    3.7980  
9796          Art           BIC Brite Liner Highlighters, Chisel Tip   10.3680  
9797       Phones                                        GE 30524EE4  235.1880  
9798       Phones           Anker 24W Portable Micro USB Car Charger   26.3760  
9799  Accessories                SanDisk Cruzer 4 GB USB Flash Drive   10.3840  

[9800 rows x 19 columns]

# Customer and Order Analysis <a id="customer_and_order_analysis"></a>

In [7]:
query2="""
SELECT 
    Customer_ID,
    Customer_Name,
    COUNT(Product_ID) AS Sales_Volume
FROM 
    retail_superstore 
GROUP BY 
    Customer_ID,
    Customer_Name
ORDER BY 
    Sales_Volume DESC
LIMIT 10
"""
result = pd.read_sql_query(query2, conn)
result

Customer_ID        Customer_Name  Sales_Volume
0    WB-21850        William Brown            35
1    MA-17560         Matt Abelman            34
2    PP-18955           Paul Prost            34
3    JL-15835             John Lee            33
4    CK-12205  Chloris Kastensmidt            32
5    JD-15895     Jonathan Doherty            32
6    SV-20365          Seth Vernon            32
7    AP-10915       Arthur Prichep            31
8    EP-13915           Emily Phan            31
9    ZC-21910     Zuschuss Carroll            31

- The Customer with highest Sales Volume is William Brown (Customer ID= WB-21850 and Sales_Volume=35)

2.What are the different customer segments, and how do they differ in purchasing behavior?

In [8]:
query3="""
SELECT 
    Segment, 
    COUNT(DISTINCT Customer_ID) AS Number_of_Customers,
    COUNT(Order_ID) AS Total_Orders,
    SUM(Sales) AS Total_Sales,
    AVG(Sales) AS Average_Order_Value,
    SUM(Sales) / COUNT(DISTINCT Customer_ID) AS Average_Sales_Per_Customer
FROM 
    retail_superstore
GROUP BY 
    Segment
ORDER BY 
    Total_Sales DESC;
"""
result = pd.read_sql_query(query3, conn)
result

Segment  Number_of_Customers  Total_Orders   Total_Sales  \
0     Consumer                  409          5101  1.148061e+06   
1    Corporate                  236          2953  6.884941e+05   
2  Home Office                  148          1746  4.249822e+05   

   Average_Order_Value  Average_Sales_Per_Customer  
0           225.065777                 2806.993963  
1           233.150720                 2917.347775  
2           243.403309                 2871.501195

# Insights
- Number Of Customers in Consumer Segment( 409 Distinct Customer) Leading to their highest total sales( 114061 USD)
- Consumer segment generates the highest total sales and revenue.


3.What are the average order values and order frequencies by customer?

In [9]:
query3="""
SELECT 
    Customer_ID,
    Customer_Name,
    COUNT(Order_ID) AS Order_Frequency,
    SUM(Sales) AS Total_Sales,
    AVG(Sales) AS Average_Order_Value
FROM 
    retail_superstore
GROUP BY 
    Customer_ID, Customer_Name
ORDER BY 
    Total_Sales DESC;

"""
result = pd.read_sql_query(query3, conn)
result

Customer_ID    Customer_Name  Order_Frequency  Total_Sales  \
0      SM-20320      Sean Miller               15    25043.050   
1      TC-20980     Tamara Chand               12    19052.218   
2      RB-19360     Raymond Buch               18    15117.339   
3      TA-21385     Tom Ashbrook               10    14595.620   
4      AB-10105    Adrian Barton               20    14473.571   
..          ...              ...              ...          ...   
788    MG-18205  Mitch Gastineau                2       16.739   
789    CJ-11875     Carl Jackson                1       16.520   
790    SC-20845       Sung Chung                1       14.112   
791    LD-16855     Lela Donovan                1        5.304   
792    TS-21085    Thais Sissman                2        4.833   

     Average_Order_Value  
0            1669.536667  
1            1587.684833  
2             839.852167  
3            1459.562000  
4             723.678550  
..                   ...  
788             8.369500  
789            16.520000  
790            14.112000  
791             5.304000  
792             2.416500  

[793 rows x 5 columns]

# Insights
- Order Frequency: How frequently each customer places orders.
- Average Order Value: The typical value of each order for a customer, which gives insight into their purchasing habits and potential value to the business.


# Product Performance <a id="product_performance"></a>

1.Which are the top-selling products, categories, and sub-categories?

In [10]:
query4="""
SELECT 
    Product_ID,
    Product_Name,
    SUM(Sales) AS Total_Sales
FROM 
    retail_superstore
GROUP BY
    product_Name
ORDER BY Total_Sales DESC
LIMIT 10
"""
result = pd.read_sql_query(query4, conn)
result

Product_ID                                       Product_Name  \
0  TEC-CO-10004722              Canon imageCLASS 2200 Advanced Copier   
1  OFF-BI-10003527  Fellowes PB500 Electric Punch Plastic Comb Bin...   
2  TEC-MA-10002412  Cisco TelePresence System EX90 Videoconferenci...   
3  FUR-CH-10002024       HON 5400 Series Task Chairs for Big and Tall   
4  OFF-BI-10001359         GBC DocuBind TL300 Electric Binding System   
5  OFF-BI-10000545   GBC Ibimaster 500 Manual ProClick Binding System   
6  TEC-CO-10001449               Hewlett Packard LaserJet 3310 Copier   
7  TEC-MA-10001127  HP Designjet T520 Inkjet Large Format Printer ...   
8  OFF-BI-10004995          GBC DocuBind P400 Electric Binding System   
9  OFF-SU-10000151        High Speed Automatic Electric Letter Opener   

   Total_Sales  
0    61599.824  
1    27453.384  
2    22638.480  
3    21870.576  
4    19823.479  
5    19024.500  
6    18839.686  
7    18374.895  
8    17965.068  
9    17030.312

In [11]:
query5="""
SELECT 
    Category,
    COUNT(Product_ID) AS product_sold_count,
    SUM(Sales) AS Total_Sales
FROM 
    retail_superstore
GROUP BY
    Category
ORDER BY Total_Sales DESC,product_sold_count DESC
LIMIT 10
"""
result = pd.read_sql_query(query5, conn)
result

Category  product_sold_count  Total_Sales
0       Technology                1813  827455.8730
1        Furniture                2078  728658.5757
2  Office Supplies                5909  705422.3340

In [12]:
query6="""
SELECT 
    Sub_Category,
    COUNT(Product_ID) AS product_sold_count,
    SUM(Sales) AS Total_Sales
FROM 
    retail_superstore
GROUP BY
    Sub_Category
ORDER BY Total_sales DESC,product_sold_count DESC
"""
result = pd.read_sql_query(query6, conn)
result

Sub_Category  product_sold_count  Total_Sales
0        Phones                 876  327782.4480
1        Chairs                 607  322822.7310
2       Storage                 832  219343.3920
3        Tables                 314  202810.6280
4       Binders                1492  200028.7850
5      Machines                 115  189238.6310
6   Accessories                 756  164186.7000
7       Copiers                  66  146248.0940
8     Bookcases                 226  113813.1987
9    Appliances                 459  104618.4030
10  Furnishings                 931   89212.0180
11        Paper                1338   76828.3040
12     Supplies                 184   46420.3080
13          Art                 785   26705.4100
14    Envelopes                 248   16128.0460
15       Labels                 357   12347.7260
16    Fasteners                 214    3001.9600

# Insights
- The top-selling product is Canon imageCLASS 2200 Advanced Copier	
- The top-selling category is Technology
- The  top-selling sub-category is Phones

2.Which products or categories drive the most revenue?

In [13]:
query7="""
SELECT
    Product_Name,
    COUNT(product_id)*AVG(Sales) AS Revenue
FROM
    retail_superstore
GROUP BY 
    Product_Name
ORDER BY
    revenue DESC
LIMIT 10
"""
result = pd.read_sql_query(query7, conn)
result

Product_Name    Revenue
0              Canon imageCLASS 2200 Advanced Copier  61599.824
1  Fellowes PB500 Electric Punch Plastic Comb Bin...  27453.384
2  Cisco TelePresence System EX90 Videoconferenci...  22638.480
3       HON 5400 Series Task Chairs for Big and Tall  21870.576
4         GBC DocuBind TL300 Electric Binding System  19823.479
5   GBC Ibimaster 500 Manual ProClick Binding System  19024.500
6               Hewlett Packard LaserJet 3310 Copier  18839.686
7  HP Designjet T520 Inkjet Large Format Printer ...  18374.895
8          GBC DocuBind P400 Electric Binding System  17965.068
9        High Speed Automatic Electric Letter Opener  17030.312

In [14]:
query8="""
SELECT
    Category,
    COUNT(product_id)*AVG(Sales) AS Revenue
FROM
    retail_superstore
GROUP BY 
    Category
ORDER BY
    revenue DESC
LIMIT 10
"""
result = pd.read_sql_query(query8, conn)
result

Category      Revenue
0       Technology  827455.8730
1        Furniture  728658.5757
2  Office Supplies  705422.3340

3.Are there specific products or categories that perform better in certain regions or states?

In [15]:

query9="""
SELECT 
    State, 
    Category, 
    Total_Sales
FROM (
    SELECT 
        State, 
        Category, 
        SUM(Sales) AS Total_Sales,
        ROW_NUMBER() OVER (PARTITION BY State ORDER BY SUM(Sales) DESC) AS Sales_Rank
    FROM 
        retail_superstore
    GROUP BY 
        State, Category
) AS RankedSales
WHERE 
    Sales_Rank = 1;
"""
result = pd.read_sql_query(query9, conn)
result

State         Category  Total_Sales
0                Alabama       Technology     8969.080
1                Arizona        Furniture    13525.291
2               Arkansas  Office Supplies     4565.330
3             California       Technology   154684.180
4               Colorado        Furniture    13220.285
5            Connecticut  Office Supplies     5418.340
6               Delaware       Technology    14562.220
7   District of Columbia       Technology     1379.920
8                Florida       Technology    46968.036
9                Georgia  Office Supplies    26397.780
10                 Idaho        Furniture     2595.482
11              Illinois       Technology    31637.881
12               Indiana       Technology    25959.670
13                  Iowa        Furniture     2627.400
14                Kansas  Office Supplies     1954.150
15              Kentucky       Technology    12552.640
16             Louisiana  Office Supplies     3423.160
17                 Maine       Technology      761.250
18              Maryland  Office Supplies    10390.230
19         Massachusetts  Office Supplies    11988.740
20              Michigan  Office Supplies    37688.199
21             Minnesota  Office Supplies    19406.540
22           Mississippi        Furniture     4317.850
23              Missouri  Office Supplies    12182.180
24               Montana       Technology     3662.934
25              Nebraska       Technology     3285.740
26                Nevada  Office Supplies     6956.924
27         New Hampshire       Technology     3636.800
28            New Jersey       Technology    14501.160
29            New Mexico        Furniture     1701.412
30              New York       Technology   126902.690
31        North Carolina       Technology    26083.119
32          North Dakota  Office Supplies      919.910
33                  Ohio       Technology    34550.548
34              Oklahoma        Furniture     8284.100
35                Oregon        Furniture     6338.130
36          Pennsylvania       Technology    42064.069
37          Rhode Island       Technology    10474.410
38        South Carolina  Office Supplies     3811.840
39          South Dakota  Office Supplies      597.720
40             Tennessee        Furniture    13506.732
41                 Texas       Technology    64656.272
42                  Utah        Furniture     4822.350
43               Vermont        Furniture     5120.100
44              Virginia        Furniture    25321.950
45            Washington       Technology    50536.710
46         West Virginia        Furniture      673.344
47             Wisconsin        Furniture    16757.850
48               Wyoming        Furniture     1603.136

In [16]:
query12="""
SELECT 
    Region, 
    Category, 
    Total_Sales
FROM (
    SELECT 
        Region, 
        Category, 
        SUM(Sales) AS Total_Sales,
        ROW_NUMBER() OVER (PARTITION BY Region ORDER BY SUM(Sales) DESC) AS Sales_Rank
    FROM 
        retail_superstore
    GROUP BY 
        Region, Category
) AS RankedSales
WHERE 
    Sales_Rank = 1;

"""
result = pd.read_sql_query(query12, conn)
result

Region    Category  Total_Sales
0  Central  Technology   168739.208
1     East  Technology   263116.527
2    South  Technology   148195.208
3     West  Technology   247404.930

In [17]:
query13="""
SELECT 
    Region, 
    Product_Name, 
    Total_Sales
FROM (
    SELECT 
        Region, 
        Product_Name, 
        SUM(Sales) AS Total_Sales,
        ROW_NUMBER() OVER (PARTITION BY Region ORDER BY SUM(Sales) DESC) AS Sales_Rank
    FROM 
        retail_superstore
    GROUP BY 
        Region, Product_Name
) AS RankedSales
WHERE 
    Sales_Rank = 1;

"""
result = pd.read_sql_query(query13, conn)
result

Region                                       Product_Name  Total_Sales
0  Central              Canon imageCLASS 2200 Advanced Copier    17499.950
1     East              Canon imageCLASS 2200 Advanced Copier    30099.914
2    South  Cisco TelePresence System EX90 Videoconferenci...    22638.480
3     West              Canon imageCLASS 2200 Advanced Copier    13999.960

In [18]:
query13="""
SELECT 
    State, 
    Product_Name, 
    Total_Sales
FROM (
    SELECT 
        State, 
        Product_Name, 
        SUM(Sales) AS Total_Sales,
        ROW_NUMBER() OVER (PARTITION BY State ORDER BY SUM(Sales) DESC) AS Sales_Rank
    FROM 
        retail_superstore
    GROUP BY 
        State, Product_Name
) AS RankedSales
WHERE 
    Sales_Rank = 1;

"""
result = pd.read_sql_query(query13, conn)
result

State                                       Product_Name  \
0                Alabama  Hewlett-Packard Deskjet 3050a All-in-One Color...   
1                Arizona                             Samsung Galaxy S4 Mini   
2               Arkansas         GBC DocuBind TL300 Electric Binding System   
3             California        High Speed Automatic Electric Letter Opener   
4               Colorado          Lexmark MX611dhe Monochrome Laser Printer   
5            Connecticut                             Hon GuestStacker Chair   
6               Delaware              Canon imageCLASS 2200 Advanced Copier   
7   District of Columbia                     Cisco SPA525G2 5-Line IP Phone   
8                Florida  Cisco TelePresence System EX90 Videoconferenci...   
9                Georgia  Fellowes PB500 Electric Punch Plastic Comb Bin...   
10                 Idaho         Bretford Rectangular Conference Table Tops   
11              Illinois                                  Samsung Galaxy S4   
12               Indiana              Canon imageCLASS 2200 Advanced Copier   
13                  Iowa                           Hon Olson Stacker Stools   
14                Kansas                   Standard Rollaway File with Lock   
15              Kentucky                 Cisco 9971 IP Video Phone Charcoal   
16             Louisiana  Martin Yale Chadless Opener Electric Letter Op...   
17                 Maine   Wi-Ex zBoost YX540 Cellular Phone Signal Booster   
18              Maryland  Fellowes PB500 Electric Punch Plastic Comb Bin...   
19         Massachusetts                Hoover Upright Vacuum With Dirt Cup   
20              Michigan   GBC Ibimaster 500 Manual ProClick Binding System   
21             Minnesota               Ibico EPK-21 Electric Binding System   
22           Mississippi  Office Impressions End Table, 20-1/2"H x 24"W ...   
23              Missouri                 Canon PC1060 Personal Laser Copier   
24               Montana                      Canon Image Class D660 Copier   
25              Nebraska            ClearOne CHATAttach 160 - speaker phone   
26                Nevada               Ibico EPK-21 Electric Binding System   
27         New Hampshire                      Logitech diNovo Edge Keyboard   
28            New Jersey   3D Systems Cube Printer, 2nd Generation, Magenta   
29            New Mexico                           SAFCO Arco Folding Chair   
30              New York              Canon imageCLASS 2200 Advanced Copier   
31        North Carolina          Cubify CubeX 3D Printer Triple Head Print   
32          North Dakota                    Fellowes High-Stak Drawer Files   
33                  Ohio          Cubify CubeX 3D Printer Double Head Print   
34              Oklahoma      Global Leather and Oak Executive Chair, Black   
35                Oregon         GuestStacker Chair with Chrome Finish Legs   
36          Pennsylvania              Canon imageCLASS 2200 Advanced Copier   
37          Rhode Island               Hewlett Packard LaserJet 3310 Copier   
38        South Carolina                    Global Comet Stacking Arm Chair   
39          South Dakota  1.7 Cubic Foot Compact "Cube" Office Refrigera...   
40             Tennessee   GBC Ibimaster 500 Manual ProClick Binding System   
41                 Texas          Lexmark MX611dhe Monochrome Laser Printer   
42                  Utah           Hewlett Packard 310 Color Digital Copier   
43               Vermont  Riverside Palais Royal Lawyers Bookcase, Royal...   
44              Virginia  HP Designjet T520 Inkjet Large Format Printer ...   
45            Washington              Canon imageCLASS 2200 Advanced Copier   
46         West Virginia  Chromcraft 48" x 96" Racetrack Double Pedestal...   
47             Wisconsin  Office Star - Professional Matrix Back Chair w...   
48               Wyoming      Global Troy Executive Leather Low-Back Tilter   

    Total_Sales  
0      3040.000  
1      1879.960  
2      1793.98

# Geographic Analysis <a id="geographic_analysis"></a>

What regions or cities have the highest sales volume?

In [19]:
query10="""
SELECT 
    Region,
    SUM(Sales) AS "Total Sales"
FROM
    retail_superstore
GROUP BY 
    Region
ORDER BY 
    "Total Sales" DESC
LIMIT 10
"""
result = pd.read_sql_query(query10, conn)
result

Region  Total Sales
0     West  710219.6845
1     East  669518.7260
2  Central  492646.9132
3    South  389151.4590

In [20]:
query11="""
SELECT 
    City,
    SUM(Sales) AS "Total Sales"
FROM
    retail_superstore
GROUP BY 
    City
ORDER BY 
    "Total Sales" DESC
LIMIT 10
"""
result = pd.read_sql_query(query11, conn)
result

City  Total Sales
0  New York City  252462.5470
1    Los Angeles  173420.1810
2        Seattle  116106.3220
3  San Francisco  109041.1200
4   Philadelphia  108841.7490
5        Houston   63956.1428
6        Chicago   47820.1330
7      San Diego   47521.0290
8   Jacksonville   44713.1830
9        Detroit   42446.9440

# Insights
- West Region has highest sales volume.
- New York City has highest Sales Volume.

Are there regional differences in product preferences?

In [21]:
query14="""
SELECT 
    Region, 
    Product_ID, 
    Product_Name, 
    Total_Sales
FROM (
    SELECT 
        Region, 
        Product_ID, 
        Product_Name, 
        SUM(Sales) AS Total_Sales,
        ROW_NUMBER() OVER (PARTITION BY Region ORDER BY SUM(Sales) DESC) AS Sales_Rank
    FROM 
        retail_superstore
    GROUP BY 
        Region, Product_ID, Product_Name
) AS RankedProducts
WHERE 
    Sales_Rank <= 5  -- Selecting top 5 products in each region
ORDER BY 
    Region, Sales_Rank;

"""
result = pd.read_sql_query(query14, conn)
result

Region       Product_ID  \
0   Central  TEC-CO-10004722   
1   Central  TEC-MA-10000822   
2   Central  OFF-BI-10001120   
3   Central  OFF-BI-10000545   
4   Central  OFF-BI-10004995   
5      East  TEC-CO-10004722   
6      East  TEC-MA-10001047   
7      East  FUR-BO-10004834   
8      East  OFF-BI-10001359   
9      East  TEC-CO-10001449   
10    South  TEC-MA-10002412   
11    South  TEC-MA-10001127   
12    South  OFF-BI-10001359   
13    South  TEC-MA-10004125   
14    South  OFF-BI-10003527   
15     West  TEC-CO-10004722   
16     West  OFF-SU-10000151   
17     West  FUR-CH-10001215   
18     West  OFF-BI-10003527   
19     West  FUR-CH-10003973   

                                         Product_Name  Total_Sales  
0               Canon imageCLASS 2200 Advanced Copier    17499.950  
1           Lexmark MX611dhe Monochrome Laser Printer    14279.916  
2                Ibico EPK-21 Electric Binding System    11339.940  
3    GBC Ibimaster 500 Manual ProClick Binding System    10653.720  
4           GBC DocuBind P400 Electric Binding System     8710.336  
5               Canon imageCLASS 2200 Advanced Copier    30099.914  
6    3D Systems Cube Printer, 2nd Generation, Magenta    14299.890  
7   Riverside Palais Royal Lawyers Bookcase, Royal...    11717.034  
8          GBC DocuBind TL300 Electric Binding System     8790.502  
9                Hewlett Packard LaserJet 3310 Copier     8639.856  
10  Cisco TelePresence System EX90 Videoconferenci...    22638.480  
11  HP Designjet T520 Inkjet Large Format Printer ...    11374.935  
12         GBC DocuBind TL300 Electric Binding System     8342.007  
13          Cubify CubeX 3D Printer Triple Head Print     7999.980  
14  Fellowes PB500 Electric Punch Plastic Comb Bin...     7625.940  
15              Canon imageCLASS 2200 Advanced Copier    13999.960  
16        High Speed Automatic Electric Letter Opener    13100.240  
17      Global Troy Executive Leather Low-Back Tilter    10019.600  
18  Fellowes PB500 Electric Punch Plastic Comb Bin...     8134.336  
19         GuestStacker Chair with Chrome Finish Legs     8030.016

# Insights
- Discovered regional product preferences to tailor marketing and product offerings according to local demand.
- Top 5 products which has highest sales is given.

# Seasonal and Time-Based Analysis <a id="i1"></a>

What are the peak sales months or seasons?

In [22]:
query="""
SELECT 
    CASE 
        WHEN CAST(substr(Order_Date, 4, 2) AS INTEGER) IN (12, 1, 2) THEN 'Winter'
        WHEN CAST(substr(Order_Date, 4, 2) AS INTEGER) IN (3, 4, 5) THEN 'Spring'
        WHEN CAST(substr(Order_Date, 4, 2) AS INTEGER) IN (6, 7, 8) THEN 'Summer'
        WHEN CAST(substr(Order_Date, 4, 2) AS INTEGER) IN (9, 10, 11) THEN 'Fall'
    END AS Season,
    SUM(Sales) AS Total_Sales
FROM 
    retail_superstore
GROUP BY 
    Season
ORDER BY 
    Total_Sales DESC;

"""
result = pd.read_sql_query(query, conn)
result

Season  Total_Sales
0    Fall  849761.4174
1  Spring  487943.3115
2  Winter  475142.9145
3  Summer  448689.1393

In [23]:
query="""
SELECT 
    CAST(substr(Order_Date, 4, 2) AS INTEGER) AS Month,
    SUM(Sales) AS Total_Sales
FROM 
    retail_superstore
GROUP BY 
    Month
ORDER BY 
    Total_Sales DESC;
"""
result = pd.read_sql_query(query, conn)
result

Month  Total_Sales
0      11  350161.7110
1      12  321480.1695
2       9  300103.4117
3      10  199496.2947
4       3  197573.5872
5       8  157315.9270
6       5  154086.7237
7       6  145837.5233
8       7  145535.6890
9       4  136283.0006
10      1   94291.6296
11      2   59371.1154

# Insights
- At Fall, Total Sales is highest
- At November , Total Sales is highest

In [24]:
query="""
SELECT 
    Ship_Mode,
    AVG(julianday(
        substr(Ship_Date, 7, 4) || '-' || substr(Ship_Date, 4, 2) || '-' || substr(Ship_Date, 1, 2)
    ) - julianday(
        substr(Order_Date, 7, 4) || '-' || substr(Order_Date, 4, 2) || '-' || substr(Order_Date, 1, 2)
    )) AS Avg_Delivery_Time
FROM 
    retail_superstore
WHERE 
    Ship_Date IS NOT NULL AND Order_Date IS NOT NULL
GROUP BY 
    Ship_Mode
ORDER BY 
    Avg_Delivery_Time ASC;

"""
result = pd.read_sql_query(query, conn)
result

Ship_Mode  Avg_Delivery_Time
0        Same Day           0.044610
1     First Class           2.179214
2    Second Class           3.249211
3  Standard Class           5.008363

# Business Insights 
- Customers from consumer segment generates 50.84 % revenue.
- products in technology category generates  36.58 % revenue.
- West region generates 31.40 % revenue while south region generates 11.2 % revenue.
# Recommendations
- Targeting customers in the consumer segment is advantageous.
- Maintaining an inventory of products in the technology category is advantageous.
- It is essential to concentrate on the southern region in order to boost sales.